src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/EventLog.cs

```diff
@@ -692,7 +693,23 @@ internal static RegistryKey GetEventLogRegKey(string machine, bool writable)
 
         internal static string GetDllPath(string machineName)
         {
-            return Path.Combine(NetFrameworkUtils.GetLatestBuildDllDirectory(machineName), DllName);
+            string dllPath = Path.Combine(NetFrameworkUtils.GetLatestBuildDllDirectory(machineName), DllName);
```

Comment:
Should we always use the local file now and not depend on the .NET Framework file at all?

Reply:
I considered this and wanted to get folks opinion:

> I also chose to keep the .NETFramework path as the default if the file is present, but would be open to hearing folks opinion on deleting this part of the code.

One case that folks might care about is an application that doesn't have a stable path.  Once an event source is registered, it won't be re-registered.  If an app had a pattern of inconsistent base-path and was registering this the first time, that would be broken after the app was moved.   I'm not sure how likely this would be since you need to elevate to register the source.  I suppose one scenario might be enterprise deployed PS scripts or applications.

Another concern is that we might want to service this for 5.0 since we officially support Windows ARM64.  Making this type of change might be too disruptive in servicing.

Reply:
@Anipik @jkotas what do you think.  Am I being to paranoid about this case?  I'd love to delete the .NETFramework dependency.  I suppose we could make that change and tell folks that were broken to manually change the registry, or call the API passing the .NETFramework DLL path?  Maybe make that change in 6.0, but make the safer change in 5.0 servicing?

Reply:
I think it is fine to keep the .NET Framework path

Reply:
Seems fine to me, we can always remove it later.

Reply:
agreed, if need be we can always remove it later